src/webgl/p5.RendererGL.js

```diff
@@ -1113,4 +1122,93 @@ p5.prototype._assert3d = function(name) {
     );
 };
 
+// Initializing GLU Tesselator
+var tessy = (function initTesselator() {
+  // function called for each vertex of tesselator output
+  function vertexCallback(data, polyVertArray) {
+    polyVertArray[polyVertArray.length] = data[0];
+    polyVertArray[polyVertArray.length] = data[1];
+    polyVertArray[polyVertArray.length] = data[2];
+  }
+  function begincallback(type) {
+    if (type !== libtess.primitiveType.GL_TRIANGLES) {
+      console.log('expected TRIANGLES but got type: ' + type);
+    }
+  }
+  function errorcallback(errno) {
+    console.log('error callback');
+    console.log('error number: ' + errno);
+  }
+  // callback for when segments intersect and must be split
+  function combinecallback(coords, data, weight) {
+    return [coords[0], coords[1], coords[2]];
+  }
+  function edgeCallback(flag) {
+    // don't really care about the flag, but need no-strip/no-fan behavior
+  }
+
+  var tessy = new libtess.GluTesselator();
+  tessy.gluTessCallback(libtess.gluEnum.GLU_TESS_VERTEX_DATA, vertexCallback);
+  tessy.gluTessCallback(libtess.gluEnum.GLU_TESS_BEGIN, begincallback);
+  tessy.gluTessCallback(libtess.gluEnum.GLU_TESS_ERROR, errorcallback);
+  tessy.gluTessCallback(libtess.gluEnum.GLU_TESS_COMBINE, combinecallback);
+  tessy.gluTessCallback(libtess.gluEnum.GLU_TESS_EDGE_FLAG, edgeCallback);
+
+  return tessy;
+})();
+
+p5.RendererGL.prototype._triangulate = function(contours) {
+  // libtess will take 3d verts and flatten to a plane for tesselation
+  // since only doing 2d tesselation here, provide z=1 normal to skip
+  // iterating over verts only to get the same answer.
+  // comment out to test normal-generation code
+  tessy.gluTessNormal(0, 0, 1);
+
+  var triangleVerts = [];
+  tessy.gluTessBeginPolygon(triangleVerts);
+
+  for (var i = 0; i < contours.length; i++) {
+    tessy.gluTessBeginContour();
+    var contour = contours[i];
+    for (var j = 0; j < contour.length; j += 3) {
+      var coords = [contour[j], contour[j + 1], contour[j + 2]];
+      tessy.gluTessVertex(coords, coords);
+    }
+    tessy.gluTessEndContour();
+  }
+
+  // finish polygon
+  tessy.gluTessEndPolygon();
+
+  return triangleVerts;
+};
+
+// function to calculate BezierVertex Coefficients
+p5.RendererGL.prototype._bezierCoefficients = function(t) {
+  var t2 = t * t;
+  var t3 = t2 * t;
+  var mt = 1 - t;
+  var mt2 = mt * mt;
+  var mt3 = mt2 * mt;
+  return [mt3, 3 * mt2 * t, 3 * mt * t2, t3];
+};
+
+// function to calculate QuadraticVertex Coefficients
+p5.RendererGL.prototype._quadraticCoefficients = function(t) {
+  var t2 = t * t;
+  var mt = 1 - t;
+  var mt2 = mt * mt;
+  return [mt2, 2 * mt * t, t2];
+};
+
+// function to convert Bezier coordinates to Catmull Rom Splines
+p5.RendererGL.prototype.Bezier2Catmull = function(w) {
```

Comment:
I would keep with the underscore camelCase convention here as in `_bezierToCatmull`